File: app/src/main/java/com/sabanciuniv/todoapp/model/ViewPager2Adapter.java
package com.sabanciuniv.todoapp.model;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class ViewPager2Adapter extends FragmentStateAdapter {

    private final int tabCount;

    public ViewPager2Adapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle,
                             int tabCount) {
        super(fragmentManager, lifecycle);
        this.tabCount = tabCount;
    }
    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new FragmentTodoTabs("true");
            case 1:
                return new FragmentTodoTabs("yes");
            case 2:
                return new FragmentTodoTabs("false");
            default:
                return new FragmentTodoTabs("no");
        }
    }

    @Override
    public int getItemCount() {
        return tabCount;
    }
}

File: app/src/main/java/com/sabanciuniv/todoapp/adapter/ViewPager2Adapter.kt
package com.sabanciuniv.todoapp.adapter

import androidx.datastore.core.DataStore
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.sabanciuniv.todoapp.activity.FragmentTodoTabs
import com.sabanciuniv.todoapp.fragment.FragmentNotesTab
import com.sabanciuniv.todoapp.model.FoodData

class ViewPager2Adapter(
    fragmentManager: FragmentManager, lifecycle: Lifecycle,
    private val tabCount: Int
) :
    FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> FragmentTodoTabs("false")
            1 -> FragmentNotesTab()
            2 -> FragmentTodoTabs("true")
            else -> FragmentTodoTabs("false")
        }
    }

    override fun getItemCount(): Int {
        return tabCount
    }
}